scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+      val matchedFiles = searchFiles.filter {
+        it.name !in prohibitedContentObj.getExemptedFileNameList()
+          && fileNameRegex.matches(it.name)
+          && File(it.toString()).bufferedReader().lineSequence()
+          .foldIndexed(initial = false) { lineNumber, acc, lineStr ->
+            val matches = prohibitedContentRegex.matches(lineStr)
+            if (matches) {
+              logProhibitedContentFailure(
+                lineNumber = lineNumber + 1,
+                errorToShow = prohibitedContentObj.getFailureMessage(),
+                filePath = RepoFile.retrieveFilePath(it, repoPath)
+              )
+            }
+            return@foldIndexed acc || matches
+          }
+      }
+      return matchedFiles.size != 0
+    }
+
+
+    /** Logs the failures for filename pattern violation.
+     *
+     * @param repoPath the path of the repo to be analyzed
+     * @param errorToShow the filename error to be logged
+     * @param matchedFiles a list of all the files which had the filenaming violation
+     * @return log the filename failures
+     */
+    private fun logProhibitedFilenameFailure(
+      repoPath: String,
+      errorToShow: String,
+      matchedFiles: List<File>
+    ) {
+      if (matchedFiles.size != 0) {
+        print("Filename pattern violation: $errorToShow\n")
+        matchedFiles.forEach {
+          print("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}\n")
+        }
+        print("\n")
+      }
+    }
+
+    /** Logs the failures for file content violation.
+     *
+     * @param lineNumberthe line number at which the failure occured
+     * @param errorToShow the failure message to be logged
+     * @param filePath the path of the file relative to the repository which failed the check
+     * @return log the prohibited content failures
+     */
+    private fun logProhibitedContentFailure(
+      lineNumber: Int,
+      errorToShow: String,
+      filePath: String) {
+      println(
+        "Prohibited content usage found on line no. $lineNumber\n" +
+          "File: [ROOT]/$filePath\n" +
+          "Failure message: $errorToShow\n"
+      )
+    }
+  }
+
+  /** helper class which contains all the file related helper methods. */
+  private class RepoFile() {
+    companion object {
+      /**
+       * Collects the paths of all the files which are needed to be checked.
+       *
+       * @param repoPath the path of the repo
+       * @param allowedDirectories a list of all the directories which needs to be checked
+       * @return all files which needs to be checked
+       */
+      fun collectSearchFiles(
+        repoPath: String,
+        allowedDirectories: List<String>)
+        : List<File> {
+        return File(repoPath).walk().filter { it ->
+          checkIfAllowedDirectory(
+            retrieveFilePath(it, repoPath),
+            allowedDirectories)
+            && it.isFile
```

Comment:
```suggestion
            allowedDirectories
          )
            && it.isFile
```

Reply:
It will be good to perform the multiple operations in multiple line, though we can perform in single go but it's not much readable. 

```
return File(repoPath).walk().filter { it ->
          checkIfAllowedDirectory(
            retrieveFilePath(it, repoPath),
            allowedDirectories
          )
            && it.isFile
        }.toList()
```
get the output of `checkIfAllowedDirectory` and put it in a val and then compare it with ` && it.isFile`. This way the function will be more readable.

Reply:
okay, done.
PTAL.